packages/ton-client/src/main.ts

```diff
@@ -26,13 +34,49 @@ const NUMBER_OF_BUFFERS = 3
 
 window.addEventListener('load', () => {
   // Prepare WebGL stuff
+  const cam = document.getElementById('camera') as HTMLVideoElement
+  let arController: any = undefined
+
+  let cameraMatrix = mat4.create()
+
+  cam.addEventListener('play', () => {
+    console.log(cam.videoWidth, cam.videoHeight)
+    ARToolkit.ARController.initWithImage(
+      cam, 
+      '/camera_para.dat', {
+        orientation: 'landscape',
+      }
+    ).then((controller: any) => { 
+      controller.setPatternDetectionMode(artoolkit.AR_MATRIX_CODE_DETECTION);
+      controller.setMatrixCodeType(artoolkit.AR_MATRIX_CODE_3x3_HAMMING63);
+      arController = controller
+      const cameraMatrixF64 = arController.getCameraMatrix()
+      arController.arglCameraViewRHf(cameraMatrixF64, cameraMatrix) 
+      cameraMatrix = mat4.clone(cameraMatrixF64)
+      console.log(cameraMatrix)
```

Comment:
remove?

Reply:
Nice catch. 
I tried a lot to get the camera projection matrix right. 
These four lines can be replaced by a single one
```
      cameraMatrix = mat4.clone(arController.getCameraMatrix())
```
Did that!